### PR TITLE
[compiler-rt] Simplify and rename of operator_new_size_type

### PR DESCRIPTION
**Pull request by arichardson**
We can rely on the compiler-provided macro __SIZE_TYPE__ for all
non-MSVC compilers and fall back to `uptr` otherwise.
I verified via https://godbolt.org/z/MW9KMjv5f that this works for MSVC
as well as GCC 4.5 Clang 3.0, so that should cover supported compilers.

While touching this also rename operator_new_size_type to usize which
makes it more obvious that this is the equivalent to size_t within
the sanitizers runtime (which I plan to use in follow-up changes).

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h [hunk 1]
```diff
@@ -191,15 +191,10 @@ typedef uptr OFF_T;
...
+#ifdef __SIZE_TYPE__
+typedef __SIZE_TYPE__ usize;
 #else
-# if defined(__s390__) && !defined(__s390x__)
-// Special case: 31-bit s390 has unsigned long as size_t.
-typedef unsigned long operator_new_size_type;
-# else
-typedef u32 operator_new_size_type;
-# endif
+typedef uptr usize;
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h:197]: will it work for s390?

**Review by vitalybuka — APPROVED** [review]

File: compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h (hunk 1 above)
Comment (arichardson) [reply compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h:197]: Looks like it's correct: https://godbolt.org/z/rjnhWeW6v

**Comment by zmodem** [comment]
This broke building with MSVC for 32-bit x86: #101998. Can we revert it? It still seems to revert cleanly, and I can't really think of a better fix than putting the old `operator_new_size_type` back.

**Comment by arichardson** [comment]
> This broke building with MSVC for 32-bit x86: #101998. Can we revert it? It still seems to revert cleanly, and I can't really think of a better fix than putting the old `operator_new_size_type` back.

What is size_t on 32-bit windows? Is it `unsigned long` or `unsigned int` Surprised it's not the same as `uptr`? I will upload a patch shortly that should hopefully fix MSVC.

**Comment by zmodem** [comment]
> What is size_t on 32-bit windows?

It seems to be `unsigned int`.

**Comment by arichardson** [comment]
> > What is size_t on 32-bit windows?
> 
> It seems to be `unsigned int`.

I just looked at this in godbolt and it seems like it should work as intended on MSVC 32-bit: https://godbolt.org/z/3KT59qKoh Do you build with any special flags that could affect this?

**Comment by arichardson** [comment]
It looks like the problem is 

#  if (SANITIZER_WORDSIZE == 64) || SANITIZER_APPLE || SANITIZER_WINDOWS, this should not be checking SANITIZER_WINDOWS, since _WIN32 uses unsigned int not unsigned long. Will upload a PR.

**Pull request merged**